File: server/daily-tasks/src/main/java/com/azatdev/dailytasks/domain/models/Task.java
package com.azatdev.dailytasks.domain.models;

import java.time.ZonedDateTime;
import java.util.UUID;

public record Task(
    Long id,
    UUID ownerId,
    String title,
    ZonedDateTime createdAt,
    ZonedDateTime updatedAt,
    Status status,
    Priority priority,
    String description
) {

    public enum Status {
        NOT_STARTED,
        IN_PROGRESS,
        COMPLETED
    }

    public enum Priority {
        LOW,
        MEDIUM,
        HIGH
    }
}

File: client/src/domain/models/Task.ts
import TaskPriority from "./TaskPriority";
import TaskStatus from "./TaskStatus";

export type TaskId = number;

interface Task {
    id: TaskId;
    title: string;
    description: string;
    status: TaskStatus;
    priority: TaskPriority | undefined;
    createdAt: Date;
    updatedAt: Date;
}

export default Task;
